src/clockcache.c

```diff
@@ -3054,14 +3054,46 @@ clockcache_print(platform_log_handle *log_handle, clockcache *cc)
 bool
 clockcache_page_valid(clockcache *cc, uint64 addr)
 {
-   if (addr % clockcache_page_size(cc) != 0)
+   if (addr % clockcache_page_size(cc) != 0) {
+      platform_error_log("%s():%d: addr=%lu, clockcache_page_size()=%lu\n",
+                         __FUNCTION__,
+                         __LINE__,
+                         addr,
+                         clockcache_page_size(cc));
       return FALSE;
+   }
    uint64 base_addr = clockcache_extent_base_addr(cc, addr);
+   bool   retval    = FALSE;
+   bool   if_fails  = FALSE;
+   int    line      = 0;
    if (addr < allocator_get_capacity(cc->al)) {
-      return base_addr != 0 && allocator_get_ref(cc->al, base_addr) != 0;
+      retval =
+         ((base_addr != 0) && (allocator_get_ref(cc->al, base_addr) != 0));
+      if_fails = TRUE;
+      line = __LINE__;
```

Comment:
This function can return `FALSE` from either side of the `if()`.

These diagnostics tease apart the piece that is failing, and print appropriate diagnostics data to simplify troubleshooting which side of these conditions failed.

Reply:
I find this very confusing.  Here's a simpler approach:

1. Restore the original control flow.
2. Move the `base_addr != 0` into the `if` condition (`if (base_addr != 0 && addr < allocator_get_capacity(cc->al)) {`.
3. In the true branch of the `if`, store the refcount in a local variable.  If it is 0, print a diagnostic. Then `return refcount != 0;`
4. In the false branch of the `if`, print a diagnostic of all the relevant variables and return FALSE.

Reply:
@rtjohnso - Reworked per your suggestion. It's much cleaner now. thanks!